src/os/bluestore/BlueFS.cc

```diff
@@ -1889,7 +1918,17 @@ void BlueFS::sync_metadata()
   flush_bdev(); // FIXME?
   _flush_and_sync_log(l);
   for (unsigned i = 0; i < to_release.size(); ++i) {
+    int r = 0;
+    if (cct->_conf->bdev_enable_discard && cct->_conf->bdev_async_discard) {
+      if (!to_release[i].empty())
+        r = bdev[i]->queue_discard(to_release[i]);
+      if (r == 0)
+        continue;
+    }
+
     for (auto p = to_release[i].begin(); p != to_release[i].end(); ++p) {
+      if (cct->_conf->bdev_enable_discard && r == 0)
+        bdev[i]->discard(p.get_start(), p.get_len());
```

Comment:
This doesn't seem right.. first, if you discard async, you do'nt want ot later discard() sync, or call release(), right?